plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithMultinodeQueryRunner.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.cache;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.hive.HiveQueryRunner;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import static io.trino.tpch.TpchTable.NATION;
+
+@Test(singleThreaded = true)
+public class TestCachingHiveMetastoreWithMultinodeQueryRunner
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return HiveQueryRunner.builder()
+                .setHiveProperties(ImmutableMap.of("hive.metastore-cache-ttl", "60m"))
+                // only so that tpch schema is created
+                .setInitialTables(ImmutableList.of(NATION))
+                // multinode with dedicated worker and coordinator
+                .setNodeCount(2)
```

Comment:
node count 2 is not necessary to reproduce original issue. actually, it would make reproduction indeterministic, so you could want to have 3 INSERTS  in the test -- actually worker count + 1, so you can have a `for` loop in the test to express the dependency between number of INSERTs and node count

Reply:
Do you mean that if there are 3 workers and a single INSERT is done then only of the worker's caches would be stale?
So the chances that the next INSERT also happens on the stale worker is not 100%?